src/solver/conv_MP_bidirectional_winograd.cpp

```diff
@@ -368,96 +349,143 @@ ConvSolution ConvMPBidirectWinograd<WinoDataH, WinoFilterH, WinoDataW, WinoFilte
-    long long int strideA = m * k * 1LL, strideB = k * n * 1LL, strideC = m * n * 1LL;
-    float alpha = 1., beta = 0.0;
-    const bool isColMajor = false, transA = true, transB = false;
-    // clang-format off
-    GemmDescriptor wino_gemm_desc{isColMajor,transA,transB,m,n,k,
-        lda,ldb,ldc,batch_count,strideA,strideB,
-        strideC,alpha,beta,params.in_data_type};
+    DEFINE_WORKSPACE_OFFSETS(wino_in.buff_info.total_byte_size,
+                             wino_wei.buff_info.total_byte_size,
+                             wino_out.buff_info.total_byte_size)
 
-    result.invoker_factory = [=](std::vector<Kernel> kernels) {
-        return [=](const Handle& handle, const AnyInvokeParams& ctx) {
-            decltype(auto) data_ctx = ctx.CastTo<conv::DataInvokeParams>();
-            const auto tensors  = data_ctx.tensors;
-            Data_t workSpace = data_ctx.workSpace;
-            float total_time    = 0;
+    InvokerFactory gemm_conv_factory;
+    std::string gemm_conv_kernel_name;
+
+    auto zeroDesc = TensorDescriptor();
+    if(isXdlops)
+    {
+        gemm_conv_kernel_name = "XDLOPS_CONV: ";
+        gemm_conv_factory     = xdlops_factory;
+    }
+    else
+    {
+        // GEMM
+        gemm_conv_kernel_name = "WRW_WINO_GEMM: ";
+
+        int m = K, k = C,
+            n   = wino_in.buff_info.size.nk * wino_in.buff_info.size.w * wino_in.buff_info.size.h;
+        int lda = m, ldb = n, ldc = n;
+        int batch_count       = wino_xform_h * wino_xform_w * group_cnt;
+        long long int strideA = m * k * 1LL, strideB = k * n * 1LL, strideC = m * n * 1LL;
+        float alpha = 1., beta = 0.0;
+        const bool isColMajor = false, transA = true, transB = false;
+        // clang-format off
+        GemmDescriptor wino_gemm_desc{isColMajor,transA,transB,m,n,k,
+            lda,ldb,ldc,batch_count,strideA,strideB,
+            strideC,alpha,beta,params.in_data_type};
+        // clang-format on
+
+        gemm_conv_factory = [=](const std::vector<Kernel>&) {
+
+            return [=](const Handle& handle, const AnyInvokeParams& ctx) {
+#if MIOPEN_USE_ROCBLAS
+                const auto data_ctx = ctx.CastTo<conv::DataInvokeParams>();
+                Data_t workSpace    = data_ctx.workSpace;
+                CallGemmStridedBatched(
+                    handle,
+                    wino_gemm_desc,
+                    workSpace,
+                    static_cast<int>(wino_wei_offset / GetTypeSize(params.in_data_type)),
+                    workSpace,
+                    static_cast<int>(wino_in_offset / GetTypeSize(params.in_data_type)),
+                    workSpace,
+                    static_cast<int>(wino_out_offset / GetTypeSize(params.in_data_type)),
+                    nullptr,
+                    false,
+                    GemmBackend_t::rocblas);
+#else
+                (void)handle;
+                (void)ctx;
+                MIOPEN_THROW(miopenStatusBadParm, "ConvMPBidirectWinograd Unsupported ");
```

Comment:
```suggestion
                MIOPEN_THROW(miopenStatusBadParm, "ConvMPBidirectWinograd is not supported");
```
And this is internal error, I think.